Sources/Variable.swift

```diff
@@ -30,10 +30,16 @@ class FilterExpression: Resolvable {
   func resolve(_ context: Context) throws -> Any? {
     let result = try variable.resolve(context)
 
-    return try filters.reduce(result) { value, filter in
+    let filteredResult = try filters.reduce(result) { value, filter in
       let arguments = try filter.1.map { try $0.resolve(context) }
       return try filter.0.invoke(value: value, arguments: arguments, context: context)
     }
+
+    if filteredResult == nil, context.environment.throwOnUnresolvedVariable {
```

Comment:
I don't fill like this is a correct place for this check. nil can be a result of filter operation itself. Also it does not cover usage of undefined variables outside of filter. 
What we need to do is to check for undefined variable in `Variable.resolve` method when we do `if current == nil { return nil }` and to throw instead of returning nil.

Reply:
Thanks for the feedback! :) I will break it down into some points:
a) "nil can be a result of filter operation itself": yes. This is the reason why I implemented this feature: the main idea behind it is that getting a nil in a template indicates a bug (either because the variable is not defined or because of a filter failing to find a matching result), and instead of silently not printing anything, I would prefer to throw.
b) "Also it does not cover usage of undefined variables outside of filter": I do not fully understand what you mean, seems to be working correctly as per the test `testUnresolvedVariable` I added. I am not very familiar with the Stencil codebase, can you give me an example of template where this will not work?
c) "What we need to do is to check for undefined variable in Variable.resolve method": I also tried that, but then the `default` filter will be useless, as we will be throwing even when a default value is provided.

Reply:
Maybe instead of `throwOnUnresolvedVariable` I should call the option `throwOnUnresolvedExpression`?

Reply:
@ilyapuchka could you have a second look at this?